src/scaraplate/automation/git.py

```diff
@@ -0,0 +1,356 @@
+                # No need for any diffs here, we just need to push that.
+                self._git.push(self.changes_branch)
+            else:
+                # The two branches have diverged, we need to compare them:
+                changes: bool = not self._git.are_one_commit_diffs_equal(
+                    local_branch, remote_branch
+                )
+
+                if changes:
+                    # We could've used force push here, but instead we delete
+                    # the branch first, because in GitLab it would also close
+                    # the existing MR (if any), and we want that instead of
+                    # silently updating the old MR.
+                    self._git.push_branch_delete(self.changes_branch)
+                    self._git.push(self.changes_branch)
+                else:
+                    logger.info(
+                        "scaraplate did update the project, but there's "
+                        "an already existing branch in remote which diff "
+                        "is equal to the just produced changes"
+                    )
+
+        # Now we should ensure that a Pull Request exists for
+        # the `self.changes_branch`, but this class is designed to be agnostic
+        # from concrete git remotes, so it should be done in a child class.
+
+    def format_commit_message(self, *, template_meta: TemplateMeta) -> str:
+        return self.commit_message_template.format(
+            # TODO retrieve path from self.clone_url and pass it here too?
+            # (but careful: that clone_url might contain credentials).
+            update_time=self.update_time,
+            scaraplate_version=scaraplate_version(),
+            template_meta=template_meta,
+        )
+
+    @classmethod
+    @contextlib.contextmanager
+    def clone(
+        cls,
+        clone_url: str,
+        *,
+        clone_ref: Optional[str] = None,
+        monorepo_inner_path: Optional[Path] = None,
+        changes_branch: str,
+        commit_author: str,
+        commit_message_template: str = (
+            "Scheduled template update ({update_time:%Y-%m-%d})\n"
+            "\n"
+            "* scaraplate version: {scaraplate_version}\n"
+            "* template commit: {template_meta.commit_url}\n"
+            "* template ref: {template_meta.head_ref}\n"
+        ),
+    ) -> Iterator["GitCloneProjectVCS"]:
+        """Provides an instance of this class by issuing ``git clone``
+        to a tempdir when entering the context manager. Returns a context
+        manager object which after ``__enter__`` returns an instance
+        of this class.
+
+        :param clone_url: Any valid ``git clone`` url.
+        :param clone_ref: Git ref to checkout after clone
+            (i.e. branch or tag name).
+        :param monorepo_inner_path: Path to the root dir of project
+            relative to the root of the repo. If None, the root of
```

Comment:
None in backquotes. There some other places around where this value is not formatted.

Reply:
Found 3 occurrences and fixed them all. Thanks.